app/src/main/java/org/mozilla/fenix/settings/creditcards/view/CreditCardEditorView.kt

```diff
@@ -49,16 +53,31 @@ class CreditCardEditorView(
         }
 
         save_button.setOnClickListener {
-            containerView.hideKeyboard()
+            saveCreditCardInfo(state)
+        }
+
+        card_number_input.text = state.cardNumber.toEditable()
+        name_on_card_input.text = state.billingName.toEditable()
+
+        bindExpiryMonthDropDown(state.expiryMonth)
+        bindExpiryYearDropDown(state.expiryYears)
+    }
+
+    /**
+     * Saves a new credit card or updates an existing one with data from user input
+     */
+    internal fun saveCreditCardInfo(state: CreditCardEditorState) {
+        containerView.hideKeyboard()
+
+        if (validateCreditCardInfo()) {
+            // TODO need to know if we're updating a number, or just round-tripping it
```

Comment:
I think we can remove this comment. We do know whether or not we're updating or adding a new credit card number.